### DISCUSSION TITLE
ASA missing/wrong URLs subproject

### DISCUSSION THREAD

**Discussion by ipaleka** [asastats/channel#126] [General]
@moloch10 pointed out to "the oldest of all" problems: [Some ASA are missing links to their project's page](https://github.com/asastats/channel/issues/125):

- - - - -

Missing
310014962 ALCH: https://www.alchemon.net/
403499324 GP: https://nexusasa.com/
559219992 OCTO: https://octorand.com/
226701642 YLDY: https://yieldly.finance/

Also, the ones that have a link do not render them consistently, e.g.
542132831 APF > URL: https://www.asaportfolio.com/
470842789 DEFLY > URL: defly.blockshake.io
287867876 OPUL > URL: https://opulous.org/
511484048 STKE > URL: [www.algostake.org](http://www.algostake.org/)
329110405 Tacos > URL: Dorastacos.com

- - - - -

This is probably the oldest "issue" we have got:

by ipaleka in [Discord](https://discord.com/channels/906917846754418770/909142143254269972/909159325036060702) on 11/13/2021:

- - - - -

@GamingToups and @Mikeyy raised an issue of wrong URLs in ASA initial configuration transaction.
 
https://discord.com/channels/906917846754418770/908054330265960478/908369331308601448
https://discord.com/channels/906917846754418770/908054330265960478/908762373299507200

Probably the optimal solution for that would be a repository on its own, created under ASA Stats GitHub organization https://github.com/asastats .

The best explanation is by example, so here's a link to Tinyman's repo for ASA icons, and we could follow the similar principles, only for URLs instead of icons: https://github.com/tinymanorg/asa-list.

Everyone, any thoughts on this? What should be the name of such repository (the URL will be https://github.com/asastats/reponame) ? 

- - - - -

We should discuss of the subproject's name and structure.
I suppose structure should follow https://github.com/tinymanorg/asa-list, so an updated URL should go in the form {ASANAME}-{ASAID} for directory, and with just a single file url.txt inside each directory with the updated URL as the only content.

- - - - -

https://trello.com/c/NkxPXvKy/19-github-subproject-for-updated-urls-needs-a-contributor

- - - - -

And a bunch of other comments pointing to the same resolution...

So, the solution is defined, we just need someone who will start with it.

**Comment by moloch10** [comment]
I don't mind starting this up; can we discuss in Discord.

**Reply by ipaleka** [reply]
> I don't mind starting this up; can we discuss in Discord.

I'd copy/paste any related and valuable discussion here anyway.

**Comment by AlgoRhythMatic** [comment]
Just a quick comment that this is great and offers a point of utility and brand exposure (even if not monetized) for other projects to also use this repository - similar to usage from Tinyman's efforts per https://github.com/tinymanorg/asa-list#using-in-your-project section.

**Comment by wxivd** [comment]
**Some other missing URL**
391379500 DPANDA: https://dopepanda.org/
352658929 Doggo: https://www.algodoggo.com/
142838028 FAME: https://algofame.org/
388069621 Neko: https://www.algoneko.net/

> So, the solution is defined, we just need someone who will start with it.

I am trying to think if this solution can be optimised to serve other purposes of future developments. Ideas revolve around two use-cases/ideas. 

1- A list of ASA URL would solve this problem of missing/wrong domains, therefore satisfies the objective. Potential problem is keeping it up to date which demands extra efforts from ASA projects, these efforts are essential for AMMs but the incentive to keep it updated on other projects may not be the same.

It is possible to fetch the URL from AlgoExplorer? with Neutral Toke status as minimum condition.

2- The point here is to consider the whole picture of future and potential applications, focusing on building blocks for future developments. ASA Stats Index page for ASA-list (Icon, URL, ASA) and utility / ASA profile per each asset held in wallet.

* Tinyman ASA-List [webpage](https://asa-list.tinyman.org/asa-list/) displays the ASA-list, allows search function and the download of icon in SVG/PNG.
* Vestige API offer [Free](https://free-api.vestige.fi/docs#/) version and [subscription for enterprise/custom made developments](https://tinychart.org/pricing).

**Reply by moloch10** [reply]
> 
> It is possible to fetch the URL from AlgoExplorer? with Neutral Toke status as minimum condition.
> 
@ipaleka, since you're already scraping for Coin Market Cap links, how about an additional one for AlgoExplorer? I checked about a dozen or so ASAs and they all had a URL listed.

**Reply by ipaleka** [reply]
> I am trying to think if this solution can be optimised to serve other purposes of future developments.

Yes, and we had a lot of Discord chats about that, pretty much lost in a void - I don't have time to c/p them.

> It is possible to fetch the URL from AlgoExplorer? with Neutral Toke status as minimum condition.

We should check with Algo Explorer do they offer that information to third parties.

P.S. maintaining the repo would probably be much simpler, easier, and cheaper than maintaining a scraping code.

**Reply by ipaleka** [reply]
> > It is possible to fetch the URL from AlgoExplorer? with Neutral Toke status as minimum condition.
> 
> @ipaleka, since you're already scraping for Coin Market Cap links, how about an additional one for AlgoExplorer? I checked about a dozen or so ASAs and they all had a URL listed.

Scraping is the last option when we don't have anything else to do.

**Reply by moloch10** [reply]
> P.S. maintaining the repo would probably be much simpler, easier, and cheaper than maintaining a scraping code.

> Scraping is the last option when we don't have anything else to do.

Gotcha, if that's the case, I don't mind starting this off. 

Should it follow Tinyman's model?

Directories named $symbol-$id 

Inside each directory a file named url.txt with only their URL?

**Reply by ipaleka** [reply]
Whatever suits you and/or the community the best. To connect with the engine, we'll need a distinct ID, for your example that's simple as splitting a directory name by "-" and converting the second part into an integer.

**Reply by moloch10** [reply]
> Whatever suits you and/or the community the best. To connect with the engine, we'll need a distinct ID, for your example that's simple as splitting a directory name by "-" and converting the second part into an integer.

Ok then. Can you set up the repo or let me know where to set it up please?

**Reply by ipaleka** [reply]
> > Whatever suits you and/or the community the best. To connect with the engine, we'll need a distinct ID, for your example that's simple as splitting a directory name by "-" and converting the second part into an integer.
> 
> Ok then. Can you set up the repo or let me know where to set it up please?

As soon as a discussion about the name ends here I'll create it under /asastats/reponame.

P.S. I can create it under a temporary name, but that name then will stay forever. We have waited until now, we can wait a day or two more. :)

**Comment by wxivd** [comment]
> Everyone, any thoughts on this? What should be the name of such repository (the URL will be https://github.com/asastats/reponame) ?

https://github.com/asastats/asa-ripe
ASA: Algorand Standard Asset 
RIPE: Repository for information processing and expansion

**Reply by ipaleka** [reply]
> RIPE: Repository for information processing and expansion

"Repository" is unnecessary here, it's implied that is a repository.

Also, we're ASA Stats, so pretty much everything is about ASA, so IMO "asa" is also a redundancy.

So if we should go with an abbreviation I'm for *aipe*.

**Reply by ipaleka** [reply]
Tho we should consider a name that explains what this actually is for: fixing other people's mess! Algorand blockchain has provided everything that we need on-chain, but people act like people and make mistakes like creating their ASA without a proper URL.

**Reply by AlgoRhythMatic** [reply]
Some light-hearted url<suffix> options:
- cleaning up your mess: _urlmaid_ or _urlhousekeeping_
- insinuation of medical repairs - url combined w/ md (abbreviation for doctor): _urlmd_
- play on US toy store [Toys "R" Us](https://en.wikipedia.org/wiki/Toys_%22R%22_Us) - _"R" us_ suggesting a group of people or repository of products of a specified kind: _urlsrus_
- silly robotic suffix: _urlatron_

**Reply by moloch10** [reply]
I think `url` should be kept out of the directory name.  Today we are only using this for URLs but who knows what's to come.  We might need/want to use this location for other data points that we don't even know of yet. 

I think something along the lines of  Tinyman's `asa-list` would be a good choice.

> Also, we're ASA Stats, so pretty much everything is about ASA, so IMO "asa" is also a redundancy.

I agree to an extent but let's just say we used TM's name but with ipaleka's suggestion, it would be: `https://github.com/asastats/list`
Looking at the directory name without knowing anything, one would have no idea what kind of list is present; so it kind of is and isn't redundant.

Other generic options: `asa-list`, `asa-data`, `asa-info`, `asa-411` (AlgoRhythMatic's motif)

**Reply by ipaleka** [reply]
> I think `url` should be kept out of the directory name.  Today we are only using this for URLs but who knows what's to come.  

Absolutely! Bare and adequate minimum is asset's ID, but it's a good practice to append asset's unit name too in order to get a humanized asset's representation.

I'm for `{UNIT}-{ID}` like in tinymanorg's repo.

**Reply by ipaleka** [reply]
> Other generic options: asa-list, asa-data, asa-info, asa-411 (AlgoRhythMatic's motif)

I'm for `asa-update` or `offchain-data` or `asa-offchain-data` to correctly describe what this repo is about.

**Reply by wxivd** [reply]
`offchain-data` or `asa-offchain-data`

I will go with one of there two options

**Reply by AlgoRhythMatic** [reply]
> `offchain-data`

Seems that this is the simplest option to describe exactly what is needed here.  +1

**Reply by ipaleka** [reply]
> > `offchain-data`
> 
> Seems that this is the simplest option to describe exactly what is needed here. +1

Ok then, it's created under that name! :)

https://github.com/asastats/offchain-data

@moloch10

**Reply by moloch10** [reply]
> > > `offchain-data`
> > 
> > 
> > Seems that this is the simplest option to describe exactly what is needed here. +1
> 
> Ok then, it's created under that name! :)
> 
> https://github.com/asastats/offchain-data
> 
> @moloch10

Oof, I didn't see this and this slipped my mind, apologies ... :flushed: 😔

Last thing before I start, should the link be to the project's homepage or to its dapp / place of usage of said token?

**Reply by ipaleka** [reply]
> Last thing before I start, should the link be to the project's homepage or to its dapp / place of usage of said token?

Please create only entries for the tokens listed here with the exact URL set.

Your task is to create a repository, and it's ok that you start with already confirmed entries, the rest is on the communities, ours and Algorand's.

People will suggest URLs in the repository and then you, me, and the others will confirm or deny them.

**Reply by moloch10** [reply]
Can you give me permissions to create directories please; currently, I can only create files.

I'll make directories named `{UNIT}-{ID}` and within each dir, the file url.txt

Thanks

**Reply by ipaleka** [reply]
> Can you give me permissions to create directories please; currently, I can only create files.
> 
> I'll make directories named `{UNIT}-{ID}` and within each dir, the file url.txt

You are set as a maintainer of the repo, that's almost the highest role possible.

A usual process for contributions to a repository on Github consists of:

- fork the repository by clicking that button in the upper right corner

- make changes to the code inside cloned/forked repo

- create a pull request

   - click "New pull request" button in https://github.com/asastats/offchain-data/pulls
   - click compare across forks link
   - pick your repository in the right select box
   - describe your pull request
   - create a pull request

**Reply by moloch10** [reply]
> A usual process for contributions to a repository on Github consists of:
> 
> * fork the repository by clicking that button in the upper right corner
> * make changes to the code inside cloned/forked repo
> * create a pull request
>   
>   * click "New pull request" button in https://github.com/asastats/offchain-data/pulls
>   * click compare across forks link
>   * pick your repository in the right select box
>   * describe your pull request
>   * create a pull request

Thanks for 👆

I added the ones from this discussion and will add more.

**Reply by ipaleka** [reply]
Great, good job!

**Comment by ipaleka** [comment]
@moloch10 and others:

It looks like a bit of misunderstanding has occurred about this subproject.

We should update only URLs that are missing onchain. I checked the first few alphabetically and the same value already located onchain is repeated.

https://algoexplorer.io/asset/542132831
https://algoexplorer.io/asset/393537671

P.S. I guess the main reason for this confusion is the text I copied/pasted:

"the ones that have a link do not render them consistently, e.g."

We do not care about inconsistency, just about the accuracy of data.

So:

- Yieldly's missing a URL 
   Its URL should be added
- Tacos' URL isn't consistent
  Nevertheless, when you click the presented link then the correct page is opened:
  ![tacos](https://user-images.githubusercontent.com/49662536/179002933-6d27ee0b-6c0d-4e77-8a23-88d9e4ecdc83.png)
  So we don't need Tacos entry.
- APF has got a URL
  We don't add it.

**Reply by moloch10** [reply]
Understood. I will cross check all assets' algoexplorer page and remove the dups and only keep the missing ones. 

Thanks for clarifying.

**Reply by ipaleka** [reply]
> Understood. I will cross check all assets' algoexplorer page and remove the dups and only keep the missing ones.

Please bear in mind that you have to go to the first transaction of the asset in order to get that value on AlgoExplorer - AlgoExplorer adds a custom URL on the provider's request to the asset's home page.
 
> Thanks for clarifying.

No problem, sorry for copy/paste without acknowledging that will create additional problem.

**Reply by moloch10** [reply]
> Please bear in mind that you have to go to the first transaction of the asset in order to get that value on AlgoExplorer - AlgoExplorer adds a custom URL on the provider's request to the asset's home page.
> 
Here's the initial txn for ASASTATS: https://algoexplorer.io/tx/YSKVZSM3UUTK7BUUZQKIP4AVYC4UZ27J5EX4BRK7GF677OPG6NUA
It has the URL set like you mentioned. 😀
However, checking GP's initial txn: https://algoexplorer.io/tx/T2W3VF5TOP3N3EMMHYDFW4OE5XMAUQ7A7FVZZJCIWTDQ24DSCKZQ does not include the URL. 🙁

Now GP's asset page: https://algoexplorer.io/asset/403499324 does have a link https://nexusasa.com; however, it redirects to https://app.nexusasa.com. 

A few things to clarify:
1) Can I just grab the URL from the asset page?
2) OR check the initial txn and if blank, grab the asset page's URL?
3) If the URL retrieved from either 1 or 2 is wrong, an entry should be created in this repo.

**Reply by ipaleka** [reply]
> > Please bear in mind that you have to go to the first transaction of the asset in order to get that value on AlgoExplorer - AlgoExplorer adds a custom URL on the provider's request to the asset's home page.
> 
> Here's the initial txn for ASASTATS: https://algoexplorer.io/tx/YSKVZSM3UUTK7BUUZQKIP4AVYC4UZ27J5EX4BRK7GF677OPG6NUA It has the URL set like you mentioned. grinning However, checking GP's initial txn: https://algoexplorer.io/tx/T2W3VF5TOP3N3EMMHYDFW4OE5XMAUQ7A7FVZZJCIWTDQ24DSCKZQ does not include the URL. slightly_frowning_face
> 
> Now GP's asset page: https://algoexplorer.io/asset/403499324 does have a link https://nexusasa.com; however, it redirects to https://app.nexusasa.com.
> 
> A few things to clarify:
> 
> 1. Can I just grab the URL from the asset page?
> 
> 2. OR check the initial txn and if blank, grab the asset page's URL?

Option 2. - because that's the onchain value, the one we collect from the Algorand Node instance. AlgoExplorer's asset's page (Option 1.) is a counterpart to this repo of ours you're updating - just as we're going to use that data to present updated URLs, the AlgoExplorer uses a similar database/system to update assets' pages with new/updated URL.

 > 3. If the URL retrieved from either 1 or 2 is wrong, an entry should be created in this repo.

Yep. I suggest that you start only with missing URLs like Yieldly and then that we use updates as a marketing tool in order to bring awareness to the Algorand community that changing the URLs in ASA Stats is possible in a similar way how icons updating is possible in Tinyman.
